invenio_accounts/sessions.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Session stuff."""
+
+from __future__ import absolute_import, print_function
+
+import flask
+from flask_login import user_logged_in
+from sqlalchemy.exc import IntegrityError
+from werkzeug.local import LocalProxy
+
+from invenio_accounts.models import SessionActivity
+
+_datastore = LocalProxy(lambda: flask.current_app.
+                        extensions['invenio-accounts'].datastore)
+
+
+def add_session(user_id, sid_s):
+    """Adds the session to the SessionActivity table.
+
+    Remember to save the session via `app.session_interface.save_session`
+    first.
+    """
+    sact = SessionActivity(user_id=user_id, sid_s=sid_s)
+    try:
+        _datastore.db.session.add(sact)
+        _datastore.commit()
+    except IntegrityError as error:
+        _datastore.db.session.rollback()
+
+
+def login_listener(app, user):
+    """Connect to the user_logged_in signal to populate the SessionActivity
+    table when a user logs in."""
+
+    @flask.after_this_request
+    def populate_session_activity(response):
+        """Adds the current session to the SessionActivity table."""
+        # Have to save the session so that the sid_s gets generated.
+        app.session_interface.save_session(app, flask.session, response)
+        add_session(user.id, flask.session.sid_s)
+        return response
+
+
+def logout_listener(app, user):
+    """Connect to the user_logged_out signal?"""
+
+    save_sesh_sid_s = flask.session.sid_s
```

Comment:
> I just kept the naming convention used in `flask_kvsession`

Please try to be closer to naming conventions of `inveniosoftware` projects by using easily understandable names without slang expressions. Thank you